src/System.Memory/tests/Performance/Perf.MemorySlice.cs

```diff
@@ -64,5 +64,100 @@ private static void MemoryGetSpanThenSlice(int numberOfBytes)
             }
         }
 
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000)]
+        [InlineData(1000 * 1000)]
+        private static void ReadOnlyMemory_Byte_GetSpanThenSlice(int numberOfBytes)
+        {
+            ReadOnlyMemory<byte> memory = new byte[numberOfBytes];
+            int numberOfSlices = numberOfBytes / 10 - 1;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                long iters = Benchmark.InnerIterationCount;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < iters; i++)
+                    {
+                        for (int j = 0; j < numberOfSlices; j++)
+                        {
+                            var span = memory.Span.Slice(10, 1);
+                            localInt ^= span[0];
+                        }
+                    }
+                }
+                volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000)]
+        [InlineData(1000 * 1000)]
+        private static void ReadOnlyMemory_Char_GetSpanThenSlice(int numberOfChars)
+        {
+            ReadOnlyMemory<char> memory = new char[numberOfChars];
+            int numberOfSlices = numberOfChars / 10 - 1;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                long iters = Benchmark.InnerIterationCount;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < iters; i++)
+                    {
+                        for (int j = 0; j < numberOfSlices; j++)
+                        {
+                            var span = memory.Span.Slice(10, 1);
+                            localInt ^= span[0];
+                        }
+                    }
+                }
+                volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        public static void ReadOnlyMemory_Byte_TryGetArray()
+        {
+            ReadOnlyMemory<byte> memory = new byte[1];
+            ArraySegment<byte> result;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                long iters = Benchmark.InnerIterationCount;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < iters; i++)
+                    {
+                        memory.DangerousTryGetArray(out result);
+                    }
+                }
+            }
+
+            volatileInt = result.Count;
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        public static void ReadOnlyMemory_Char_TryGetArray()
```

Comment:
Do you have the perf results from these additional tests? Anything interesting that is worth sharing?

Reply:
I'll regenerate and share tomorrow.  Nothing particularly noteworthy; it's what you'd expect.